crates/project_panel/src/project_panel.rs

```diff
@@ -1259,28 +1259,113 @@ impl ProjectPanel {
             } else {
                 None
             };
-
-            cx.spawn(|this, mut cx| async move {
+            let next_selection = self.find_next_selection_after_deletion(cx);
+            cx.spawn(|panel, mut cx| async move {
                 if let Some(answer) = answer {
                     if answer.await != Ok(0) {
                         return Result::<(), anyhow::Error>::Ok(());
                     }
                 }
                 for (entry_id, _) in file_paths {
-                    this.update(&mut cx, |this, cx| {
-                        this.project
-                            .update(cx, |project, cx| project.delete_entry(entry_id, trash, cx))
-                            .ok_or_else(|| anyhow!("no such entry"))
-                    })??
-                    .await?;
+                    if let Some(task) = panel
+                        .update(&mut cx, |panel, cx| {
+                            panel
+                                .project
+                                .update(cx, |project, cx| project.delete_entry(entry_id, trash, cx))
+                        })
+                        .ok()
```

Comment:
Thanks, no worries, I'll pick it up. Let's get this right. I like the sanitization approach, as it solves this as well as removes the need to calculate `last_marked_entry_without_marked_ancestors`. It would be just be the last marked entry, given the sanitization list is sorted by its visibility order. I need to figure out this sorting part without using `visible_entries`.

In `last_marked_entry_without_marked_ancestors` logic, I used `visible_entries` for this reason only, as `marked_entries` is just a map, and it was essential to traverse those entries in that order. But, you are right it can go over thousands easily, so I will just way to sort `marked_entries` in that order.